src/interface/swarm_container.cpp

```diff
@@ -160,6 +187,8 @@ TaskStatus SwarmContainer::Receive(BoundaryCommSubset phase) {
   for (auto &s : swarmVector_) {
     if (s->Receive(phase)) {
       success++;
+      ApplySwarmBoundaryConditions(s);
```

Comment:
physical boundaries are applied here, as part of `SwarmContainer::Receive`. Maybe this could be made a little more obvious by renaming this function `ReceiveAndApplyBoundaries` or something. In general boundaries *must* be applied after send/receives to handle periodic boundary conditions so that was my motivation for calling `ApplySwarmBoundaryConditions` here.